### PR TITLE
Add ``content: write`` rights for backporting job

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

We need it to push to the backporting branch after cherry-picking. See https://github.com/PyCQA/pylint/pull/7810\#issuecomment-1324742960

Also include change we did on the maintenance branch to not run primers.

Refs #7804, #7815 #7810

### CODE REVIEW

**Review by DanielNoord — DISMISSED** [review]
Primer changes look good!

**Review by clavedeluna — APPROVED** [review]

File: .github/workflows/backport.yml [hunk 1]
```diff
@@ -7,6 +7,7 @@ on:
 
 permissions:
   pull-requests: write
+  content: write
```
Comment (cdce8p) [inline .github/workflows/backport.yml:10]: It was worth a shot at least.

**Review by cdce8p — COMMENTED** [review]
Could you add a version comment here?
https://github.com/PyCQA/pylint/blob/f7d681b5a79e5781ab8072fe64459b199955a1f6/.github/workflows/backport.yml#L26

Similar to the one here. Dependabot will auto-update that too.
https://github.com/PyCQA/pylint/blob/f7d681b5a79e5781ab8072fe64459b199955a1f6/.github/workflows/primer_comment.yaml#L134

File: .github/workflows/backport.yml [hunk 2]
```diff
@@ -23,6 +24,6 @@ jobs:
         )
       )
     steps:
-      - uses: tibdex/backport@2e217641d82d02ba0603f46b1aeedefb258890ac
+      - uses: tibdex/backport@2e217641d82d02ba0603f46b1aeedefb258890ac # v0.4.0
```
Comment (cdce8p) [inline .github/workflows/backport.yml:27]:
```suggestion
      - uses: tibdex/backport@2e217641d82d02ba0603f46b1aeedefb258890ac # v2.0.3
```

Comment (Pierre-Sassoulas) [reply .github/workflows/backport.yml:27]: Duh, I trust you too much 😄

**Review by cdce8p — APPROVED** [review]
Permissions look fine. Haven't followed the discussion around primer runs though.
If it should only run on `main`, I guess it's also good.

**Pull request merged**